src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+/** A generator of a collection of AvailabilityTimeSlot Objects. */
+public class AvailabilityTimeSlotGenerator {
+  private static final int EARLIEST_HOUR = 8;
+  private static final int LATEST_HOUR = 19;
+  // A list of hours and minutes representing permitted time slots.
+  private static final ImmutableList<HoursAndMinutes> ALL_HOURS_AND_MINUTES = allHoursAndMinutes();
+
+  @AutoValue
+  abstract static class HoursAndMinutes {
+    abstract int hour();
+
+    abstract int minute();
+
+    static HoursAndMinutes create(int hour, int minute) {
+      return builder().setHour(hour).setMinute(minute).build();
+    }
+
+    static Builder builder() {
+      return new AutoValue_AvailabilityTimeSlotGenerator_HoursAndMinutes.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
```

Comment:
This really doesn't need a builder. Drop the builder and use `create` everywhere.

Reply:
Done